extensions/dialog/client.mjs

```diff
@@ -0,0 +1,73 @@
+// Copyright 2021 HITCON Online Contributors
+// SPDX-License-Identifier: BSD-2-Clause
+
+/**
+ * This class is the browser/client side of an extension.
+ * One instance is created for each connected player.
+ */
+class Client {
+  /**
+   * Create the client side of the extension.
+   * @constructor
+   * @param {ClientExtensionHelper} helper - An extension helper object for
+   * servicing various functionalities of the extension.
+   */
+  constructor(helper) {
+    this.helper = helper;
+    document.getElementById('c2sButton').addEventListener('click', () => {this.getNpcData()});
+  }
+
+  /**
+   * Get the NPC data from Server (npc.json)
+   */
+  getNpcData(){
+    console.log('[c2s Button] getNpcData');
+    let arg = {'npc': 'Amy'} //for test
+
+    this.helper.callC2sAPI(null, 'getSpecificNpcData', 5000, arg);
+  }
+
+  /**
+   * Show the single choice dialog to user.
+   * @param {object} arg - npc data.
+   */
+  s2c_singleChoiceDialog(arg){
```

Comment:
These should be async, and will timeout (close the dialog) or return when the user does something.